src/update_handler.c

```diff
@@ -652,6 +652,7 @@ static gboolean mount_and_run_slot_hook(const gchar *hook_name, const gchar *hoo
 
 	g_assert_nonnull(hook_name);
 	g_assert_nonnull(hook_cmd);
+	g_assert_nonnull(error);
```

Comment:
Hm, ok, I just checked the current code and verified that it was never called with a NULL error, and given the existing asserts in mount_and_run_slot_hook (which, as I said, is local to update_handler.c), added to those. Then I grepped for the 'if (error)' pattern and found other instances in the same file, though their callers are further away. Anyway, I'll push a new version using g_return_val_if_fail.

Reply:
Yes, one could argue that if we have an internal function and want to assert it will always have its error propagated one could also use `g_return_val_if_fail(error, FALSE)`.
I am not fully sure if its better to use this or the other solution in such a case as one is more consistent with the expected glib handling while the other might give us a little more 'assertion power'..